pages/redirect/humanid.tsx

```diff
@@ -0,0 +1,16 @@
+import getConfig from "next/config"
+
+export async function getServerSideProps({ query, req }) {
+    const { REDIRECT_MAIN_URL } = getConfig().serverRuntimeConfig
+    const queryParamString = new URLSearchParams(query).toString()
+    return {
+        redirect: {
+            destination: `${REDIRECT_MAIN_URL}/verification?${queryParamString}`,
+            permanent: false,
+        },
+    }
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getServerSideProps` function is correctly implemented to handle server-side rendering in Next.js. However, it's important to validate the query parameters before using them in a URL redirect. This is to prevent Open Redirect vulnerabilities where an attacker can manipulate the destination URL to lead users to malicious websites.

```diff
export async function getServerSideProps({ query, req }) {
    const { REDIRECT_MAIN_URL } = getConfig().serverRuntimeConfig
+    if (!REDIRECT_MAIN_URL) {
+        return {
+            notFound: true,
+        }
+    }
    const queryParamString = new URLSearchParams(query).toString()
+    // Validate query parameters here
    return {
        redirect: {
            destination: `${REDIRECT_MAIN_URL}/verification?${queryParamString}`,
            permanent: false,
        },
    }
}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->